WordPress/Classes/Utility/ContextManager.swift

```diff
@@ -0,0 +1,236 @@
+        }
+
+        return url.appendingPathComponent("WordPress.sqlite")
+    }
+
+    func save(_ context: NSManagedObjectContext, andWait wait: Bool, withCompletionBlock completionBlock: (() -> Void)?) {
+        let block: () -> Void = {
+            self.internalSave(context)
+            DispatchQueue.main.async {
+                completionBlock?()
+            }
+        }
+
+        /// Ensure that the `context`'s concurrency type is not `confinementConcurrencyType`, since it will crash if `perform` or `performAndWait` is called.
+        guard context.concurrencyType == .mainQueueConcurrencyType || context.concurrencyType == .privateQueueConcurrencyType else {
+            block()
+            return
+        }
+
+        if wait {
+            context.performAndWait(block)
+        } else {
+            context.perform(block)
+        }
+    }
+}
+
+// MARK: - Initialise Core Data stack
+
+private extension ContextManager {
+    static func createPersistentContainer(storeURL: URL, modelName: String) -> NSPersistentContainer {
+        guard var modelFileURL = Bundle.main.url(forResource: "WordPress", withExtension: "momd") else {
+            fatalError("Can't find WordPress.momd")
+        }
+
+        if modelName != ContextManagerModelNameCurrent {
+            modelFileURL = modelFileURL.appendingPathComponent(modelName).appendingPathExtension("mom")
+        }
+
+        guard let objectModel = NSManagedObjectModel(contentsOf: modelFileURL) else {
+            fatalError("Can't create object model named \(modelName) at \(modelFileURL)")
+        }
+
+        let startupEvent = SentryStartupEvent()
+
+        do {
+            try migrateDataModelsIfNecessary(storeURL: storeURL, objectModel: objectModel)
+        } catch {
+            DDLogError("Unable to migrate store: \(error)")
+            startupEvent.add(error: error as NSError)
+        }
+
+        let storeDescription = NSPersistentStoreDescription(url: storeURL)
+        storeDescription.shouldInferMappingModelAutomatically = true
+        storeDescription.shouldMigrateStoreAutomatically = true
+        let persistentContainer = NSPersistentContainer(name: "WordPress", managedObjectModel: objectModel)
+        persistentContainer.persistentStoreDescriptions = [storeDescription]
+        persistentContainer.loadPersistentStores { _, error in
+            guard let error else {
+                return
+            }
+
+            DDLogError("Error opening the database. \(error)\nDeleting the file and trying again");
+            startupEvent.add(error: error)
+
+            // make a backup of the old database
+            do {
+                try CoreDataIterativeMigrator.backupDatabase(at: storeURL)
+            } catch {
+                startupEvent.add(error: error)
+            }
+
+            // TODO: The original Objective C implement calls `loadPersistentStores` again here, but throws an exception regardless the result.
```

Comment:
❓ I can certainly keep [the code that calls `loadPersistentStores` again](https://github.com/wordpress-mobile/WordPress-iOS/blob/24db036f22c493ef847c9e6f4fc14d9f9fbd1144/WordPress/Classes/Utility/ContextManager.m#L160-L167), but I don't see how this additional call is necessary, did I miss anything?

Reply:
>  I don't see how this additional call is necessary

Me neither.